src/sphmultipcqueue.cpp

```diff
@@ -442,9 +447,11 @@ SPHMPMCQGetNextCompleteDirectTM (SPHMPMCQ_t queue)
 			if (__TM_is_failure_persistent (TM_buff)) {
 				/* revert to GNU TM */
 #endif
+#ifdef HAS_GNU_TM
 				__transaction_atomic {
 					entryPtr = SPHMPMCQAdvanceTail((SPHLFEntryHeader_t **)&headerBlock->qtail,stride,qlo,qhi);
 				}
+#endif
 #ifdef __HTM__
```

Comment:
No. For the `__HTM__ && HAS_GNU_TM` case, the `HAS_GNU_TM` code is a fallback. If it were an `#elif`, the fallback code would not be present.